xmtp_mls/src/client.rs

```diff
@@ -752,17 +747,8 @@ where
         &self,
         provider: &XmtpOpenMlsProvider,
     ) -> Result<(), ClientError> {
-        provider
-            .transaction_async(move |provider| {
-                let provider = &provider;
-                async {
-                    self.identity()
-                        .rotate_key_package(provider, &self.api_client)
-                        .await?;
-                    Ok::<_, IdentityError>(())
-                }
-            })
-            .await?;
+        let kp_bytes = self.identity().rotate_key_package(provider)?;
```

Comment:
Q: wdyt of renaming the function to reflect the updated logic? suggestion: `rotate_and_upload_key_pacakge`

Reply:
Love it. Just changed.